torch/csrc/jit/runtime/autodiff.cpp

```diff
@@ -841,6 +841,12 @@ Gradient differentiate(std::shared_ptr<Graph>& graph) {
   // modifies df_input_vjps (new vjps are added for temporaries)
   lambdaLiftReverse(grad_desc, rev_info);
   packReturnValuesIntoTuple(grad_desc.df);
+
+  // we have created a differentiable forward graph
+  // which will be run with tensors that have their gradients detached,
+  // so profiled types will have outdated requires_grad=True, update the
+  // requires_grad property
+  UpdateDifferentiableGraphRequiresGrad(grad_desc.f, false);
```

Comment:
Is it true for inputs/outputs of the graph as well?

Reply:
it is true but the only downstream user of this property is the profiling executor, which only looks at prim::profile nodes. i can update the. comment